### DISCUSSION TITLE
MockResponse for same URL but different methods?

### DISCUSSION THREAD

**Discussion by MCKLtech** [saloonphp/saloon#401] [Q&A]
I'm working with an API which has a /users endpoint that I wish to test with. The issue is that to search for users; I use GET but to create, I need to POST.

However, because wildcards are not supported inside a URL, I can't seem to get Saloon to work out which MockResponse to use, so it ends up failing as the wild card is always returned.

Ideally, the first should be something like /users?query=* but that doesn't seem to work.

```
Saloon::fake([
//This is for searching /users, therefore GET
            '*' => MockResponse::make(
                [
                    "items" => [],
                    "meta" => [
                        "pagination" => [
                            "current_page" => 1,
                            "next_page" => null,
                            "prev_page" => 0,
                            "total_pages" => 1,
                            "total_items" => 0,
                            "entries_info" => "1-10 of 10"
                        ]
                    ]
                ]
            ),
//This is for creation, therefore POST
            'users' => MockResponse::make([
                "id" => 123456789
              ]
              ),
        ]);
```

Therefore, is there any way to 'tell' Saloon which 'users' endpoint to use for GET and which to use for POST?

**Comment by GregoireGaonach** [comment]
Perhaps you can pass your Request class directly instead of a string and then check what you need to check?

```php
use Saloon\Enums\Method;

Saloon::fake([
    UserRequest::class => function (PendingRequest $pendingRequest) {
        return match ($pendingRequest->getMethod()) {
            Method::GET => MockResponse::make(body: '', status: 200),
            Method::POST => MockResponse::make(body: '', status: 200),
        };
    },
]);
```

**Comment by sinnbeck** [comment]
Or make a separate request class for each `GetUsersRequest` and `CreateUsersRequest`
